engine-tests/src/test/java/org/terasology/engine/world/chunks/localChunkProvider/LocalChunkProviderTest.java

```diff
@@ -81,36 +83,33 @@ public void setUp() {
                 chunkCache);
         chunkProvider.setBlockEntityRegistry(blockEntityRegistry);
         chunkProvider.setWorldEntity(worldEntity);
-        chunkProvider.setRelevanceSystem(new RelevanceSystem(chunkProvider)); // workaround. initialize loading pipeline
+        relevanceSystem = new RelevanceSystem(chunkProvider);
+        chunkProvider.setRelevanceSystem(relevanceSystem); // workaround. initialize loading pipeline
     }
 
     @AfterEach
     public void tearDown() {
         chunkProvider.shutdown();
     }
 
-    private Future<Chunk> requestCreatingOrLoadingArea(Vector3ic chunkPosition, int radius) {
-        Future<Chunk> chunkFuture = chunkProvider.createOrLoadChunk(chunkPosition);
-        BlockRegion extentsRegion = new BlockRegion(
-                chunkPosition.x() - radius, chunkPosition.y() - radius, chunkPosition.z() - radius,
-                chunkPosition.x() + radius, chunkPosition.y() + radius, chunkPosition.z() + radius);
-
-        extentsRegion.iterator().forEachRemaining(pos -> {
-            if (!pos.equals(chunkPosition)) { // remove center. we takes future for it already.
-                chunkProvider.createOrLoadChunk(pos);
-            }
-        });
-        return chunkFuture;
+    private void requestCreatingOrLoadingArea(Vector3ic chunkPosition, int radius) {
+        playerEntity = mock(EntityRef.class);
+        when(playerEntity.exists()).thenReturn(true);
+        when(playerEntity.getComponent(LocationComponent.class)).thenReturn(new LocationComponent(new Vector3f(chunkPosition)));
+        Vector3i distance = new Vector3i(radius * 2, radius * 2, radius * 2);
+        relevanceSystem.addRelevanceEntity(playerEntity, distance, null);
     }
 
-    private Future<Chunk> requestCreatingOrLoadingArea(Vector3ic chunkPosition) {
-        return requestCreatingOrLoadingArea(chunkPosition, 1);
+    private void requestCreatingOrLoadingArea(Vector3ic chunkPosition) {
+        requestCreatingOrLoadingArea(chunkPosition, 1);
     }
 
     @Test
     void testGenerateSingleChunk() throws InterruptedException, ExecutionException, TimeoutException {
         Vector3i chunkPosition = new Vector3i(0, 0, 0);
-        requestCreatingOrLoadingArea(chunkPosition).get(WAIT_CHUNK_IS_READY_IN_SECONDS, TimeUnit.SECONDS);
+        requestCreatingOrLoadingArea(chunkPosition);
+        chunkProvider.notifyRelevanceChanged();
+        Thread.sleep(WAIT_CHUNK_IS_READY_IN_SECONDS * 1000);
```

Comment:
The difference between `future.get(WAIT_…)` and `Thread.sleep(WAIT_…)` is that the `get` will take **at most** that much time to execute, but the `sleep` will take **at _least_** that much time.

Reply:
Yeah, it's annoying that they take so long now. Unfortunately there aren't any futures to wait on anymore, but I guess we could add an object to wait on in the ChunkProcessingPipeline, or use a loop.

Reply:
I was going to ask about that: Why did you remove the futures?

My guess would have been because you only call the methods involved from a thread dedicated to doing that work, so they're all blocking and don't return until the result is available. But the number of times the tests rely on `Thread.sleep` suggests that's not the case.

Reply:
You're right, there's just no need for futures currently. Everything is just scheduled for after the chunk is generated via stages. Futures are only needed in the tests, because the rest of the code just submits chunks for generation asynchronously.

Reply:
I added another object which is notified when generation finishes, which fixes the tests.